tests/ModelTest.php

```diff
@@ -722,4 +723,11 @@ public function testTruncateModel()
 
         $this->assertEquals(0, User::count());
     }
+
+    public function testGuardedModel()
+    {
+        Guarded::create(['var' => 'val']);
+
+        $this->assertEquals(1, Guarded::count());
```

Comment:
It's not related test case, i think that we should test that model Guarded doesn't set fields in guarded attribute of Model
`
$model = Guarded::create(['var' => 'val', 'foobar' => 'bar']);
$this->assertFalse(isset($model->foobar));
` 
See tests in [laravel](https://github.com/laravel/framework/blob/e351423804a816d1acd86e732a425dabddeadd3a/tests/Database/DatabaseEloquentModelTest.php#L1097-L1115)

Reply:
The error occurs when Laravel checks using an `isGuardedColumn` method. This method however doesn't perform the actual guard checks, it checks if " if the given column is a valid, guardable column" (whatever that may mean).

It's easy to add the test, but it's not strictly neccessary as the `isGuarded` check runs this statement (on Laravel 7.25.0)

```php
return $this->getGuarded() == ['*'] ||
        ! empty(preg_grep('/^'.preg_quote($key).'$/i', $this->getGuarded())) ||
        ! $this->isGuardableColumn($key);
```

([source](https://github.com/laravel/framework/blob/0d1ac3282e019ed4608b715ac4f8c57a0a042b9b/src/Illuminate/Database/Eloquent/Concerns/GuardsAttributes.php#L204-L206))

Reply:
Change test case, please, we can show that test will be broken if laravel changes behavior with guarded fields